dev/tests/api-functional/testsuite/Magento/GraphQl/Review/CreateProductReviewsTest.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\Review;
+
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Framework\Exception\AuthenticationException;
+use Magento\Framework\Registry;
+use Magento\Integration\Api\CustomerTokenServiceInterface;
+use Magento\Review\Model\ResourceModel\Review\Collection;
+use Magento\Review\Model\ResourceModel\Review\CollectionFactory as ReviewCollectionFactory;
+use Magento\Review\Model\Review;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+
+/**
+ * Test coverage for adding product reviews mutation
+ */
+class CreateProductReviewsTest extends GraphQlAbstract
+{
+    /**
+     * @var CustomerTokenServiceInterface
+     */
+    private $customerTokenService;
+
+    /**
+     * @var CustomerRepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var ReviewCollectionFactory
+     */
+    private $reviewCollectionFactory;
+
+    /**
+     * @var Registry
+     */
+    private $registry;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $objectManager = Bootstrap::getObjectManager();
+        $this->customerTokenService = $objectManager->get(CustomerTokenServiceInterface::class);
+        $this->customerRepository = $objectManager->get(CustomerRepositoryInterface::class);
+        $this->reviewCollectionFactory = $objectManager->get(ReviewCollectionFactory::class);
+        $this->registry = $objectManager->get(Registry::class);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Review/_files/visibility_to_all_ratings.php
+     * @magentoApiDataFixture Magento/GraphQl/Catalog/_files/simple_product.php
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     */
+    public function testAddProductReviewAsLoggedInCustomer()
```

Comment:
Can we merge `testAddProductReviewAsLoggedInCustomer` and `testAddProductReviewAsGuestCustomer` into a single test and use data provider? They look pretty similar, in that way we may make the test more compact. What do you think?

Reply:
Definitely makes sense. 
However looks like dataProviders run before any DataFixture, so we aren't able to load and pass the customer name as method's param, as there is no customer created yet.